infrastructure/template.yaml

```diff
@@ -146,6 +148,20 @@ Resources:
     DeletionPolicy: Delete
     UpdateReplacePolicy: Delete
 
+  QueryDeadLetterQueueAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      Namespace: "AWS/SQS"
+      MetricName: "ApproximateNumberOfMessagesVisible"
+      Dimensions:
+        - Name: "QueueName"
+          Value: !GetAtt QueryUserServicesDeadLetterQueue.QueueName
+      Statistic: "Sum"
+      Period: 300
+      EvaluationPeriods: 1
```

Comment:
Right, so these are measured in intervals then?
And we saying trigger... if there is greater than 1 message on the queue, for _one_ period of 300 seconds?

I'm guessing this could be something we could usefully apply to the SQS queues between the lambda then?
Like if things back up so we have... (arbitrarily) 100 messages on the queue for longer than 300 seconds, then we alert saying "oi your scaling rules need a look, or something is broken?"

Reply:
>Right, so these are measured in intervals then?

It's a bit complicated but essentially yes. It's designed so you can do some clever analytics for your alarms.  For these ones it just checks every 5 minutes to see if there are any messages in the queue which is pretty simple. 

We can do more complicated things like setting the `Period` to 60, `EvaluationPeriods` to 5 and `DatapointsToAlarm` to 3 (not set here - it defaults to what you set `EvaluationPeriods` to). That would mean 'trigger the alarm if we go over the threshold number for at least 3 out of the last 5 minutes in a rolling window'.

The [AWS docs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/AlarmThatSendsEmail.html#alarm-evaluation) do a good job of explaining it.